keps/sig-auth/2799-reduction-of-secret-based-service-account-token/README.md

```diff
@@ -0,0 +1,463 @@
+  auto-created secrets and extract tokens from them would fail.
+- LegacyServiceAccountTokenTracking: no impact.
+- LegacyServiceAccountTokenCleanUp: workloads that reads auto-generated
+  secrets after those secrets being considered unused by this feature and
+  removed.
+
+###### What specific metrics should inform a rollback?
+
+`serviceaccount_legacy_tokens_total`: cumulative stale service account tokens
+used.
+
+this metric is only informational and cannot deterministically tell a rollback
+is needed. there is no good way for us to detect scrapers of auto-generated
+secrets.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+no since there is not much change between a upgrade and upgrade->downgrade->upgrade.
+see section `What happens if we reenable the feature if it was previously rolled back`.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+no
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+check if there is a configmap `tracked-since` in namespace `kube-system`.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+<!--
+Pick one more of these and delete the rest.
+-->
+
+- [ ] Metrics
+  - Metric name: `serviceaccount_legacy_tokens_total`
+  - [Optional] Aggregation method:
+  - Components exposing the metric: kube-apiserver
+
+LegacyServiceAccountTokenNoAutoGeneration and LegacyServiceAccountTokenCleanUp
+might cause few workloads to fail but there is no way for us to inject metric
+in workloads to detect this.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the above SLIs?
+
+none. we expect the number recorded in the above metric going down in the long
+term.
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+none.
+
+### Dependencies
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### Does this feature depend on any specific services running in the cluster?
+
+no.
+
+### Scalability
+
+<!--
+For alpha, this section is encouraged: reviewers should consider these questions
+and attempt to answer them.
+
+For beta, this section is required: reviewers must answer these questions.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### Will enabling / using this feature result in any new API calls?
+
+up to one additional write request per day could be made to auto-generated secrets still in use.
```

Comment:
I don't see where in the KEP above this "per day" comes from.  Where is the frequency determined?

Reply:
when the legacy token is used, we check the annotation and if it was annotated at the same day, we don't write again. i updated the KEP above

Reply:
> when the legacy token is used, we check the annotation and if it was annotated at the same day, we don't write again. i updated the KEP above

thanks